src/bvals/comms/boundary_communication.cpp

```diff
@@ -98,13 +99,26 @@ TaskStatus SendBoundBufs(std::shared_ptr<MeshData<Real>> &md) {
         int idx_offset = 0;
         for (int iel = 0; iel < bnd_info(b).ntopological_elements; ++iel) {
           auto &idxer = bnd_info(b).idxer[iel];
+          const int Ni = idxer.template EndIdx<5>() - idxer.template StartIdx<5>() + 1;
           Kokkos::parallel_reduce(
-              Kokkos::TeamThreadRange<>(team_member, idxer.size()),
+              Kokkos::TeamThreadRange<>(team_member, idxer.size() / Ni),
               [&](const int idx, bool &lnon_zero) {
-                const auto [t, u, v, k, j, i] = idxer(idx);
-                const Real &val = bnd_info(b).var(iel, t, u, v, k, j, i);
-                bnd_info(b).buf(idx + idx_offset) = val;
-                lnon_zero = lnon_zero || (std::abs(val) >= threshold);
+                const auto [t, u, v, k, j, i] = idxer(idx * Ni);
+                Real *var = &bnd_info(b).var(iel, t, u, v, k, j, i);
+                Real *buf = &bnd_info(b).buf(idx * Ni + idx_offset);
+
+                Kokkos::parallel_for(Kokkos::ThreadVectorRange<>(team_member, Ni),
```

Comment:
Out of curiosity, have you tried a `TeamVectorRange` and/or the new `TeamMDRange`?

Reply:
I tried `TeamVectorRange` (as a single level of parallelism), but that seemed to be slower at least on CPU. I have not experimented with `TeamMDRange`, I think it would be interesting to try this out. 

As you pointed out, it is not clear what the origin of the performance boost from the changes is. My original thought was that it would promote vectorization on CPU when the `i`-direction, which would likely be beneficial when the `i`-direction is not `nghost`. That being said, it is also possible that there is a little benefit from the fact that we perform the indexing calculations significantly fewer times with the current setup.

Reply:
Is there any documentation for `TeamMDRange`?  Can't seem to find any.  What does it do? @lroberts36 are you going to try this, or should we just merge what we have an try this out later?

Reply:
@jdolence: I believe `TeamMDRange` should give us a Kokkos native way to have an inner loop go over multiple indices (rather than having the loop go over a single index and then calculate the multi-dimensional indices by hand as we do now). I don't necessarily see why `TeamMDRange` would give a performance boost, but that doesn't mean we shouldn't test it at some point. I would leave that for a future PR though.